hazelcast/src/main/java/com/hazelcast/jet/impl/MasterContext.java

```diff
@@ -155,6 +155,14 @@ public JobConfig jobConfig() {
         return jobRecord.getConfig();
     }
 
+    public void setJobConfig(JobConfig config) {
+        if (jobStatus != SUSPENDED) {
```

Comment:
in EE there is also `SUSPENDED_EXPORTING_SNAPSHOT` status. it should be allow to change job config in it